src/js/services/optimizely.test.js

```diff
@@ -0,0 +1,241 @@
+      await expect(client.getUserId()).resolves.toBe(
+        '00000000-0000-0000-0000-000000000000',
+      );
+    });
+
+    it('test waiting for userData to populate from null but without analytics_id', async () => {
+      glob.userData = null;
+
+      setTimeout(() => {
+        glob.userData = {
+          analytics_id: undefined,
+        };
+        window.dispatchEvent(userDataReady);
+      }, 100);
+
+      await expect(client.getUserId()).resolves.toBe(null);
+    });
+
+    it('test if userData does not exist initially and defined later with analytics_id', async () => {
+      glob.userData = null;
+      setTimeout(() => {
+        glob.userData = {
+          analytics_id: '00000000-0000-0000-0000-000000000000',
+        };
+        window.dispatchEvent(userDataReady);
+      }, 100);
+
+      await expect(client.getUserId()).resolves.toBe(
+        '00000000-0000-0000-0000-000000000000',
+      );
+    });
+  });
+
+  // Function used in TrackExperimentViewed
+  describe('isExperimentAlreadyViewed()', () => {
+    it('no experiment has been viewed before', () => {
+      expect(isExperimentAlreadyViewed(orgId, experimentKey)).toBe(null);
+    });
+    it('experiment has not been viewed before but an experiment has been viewed', () => {
+      storeExperimentParticipation(orgId, experimentKey + '456', variationName);
+      expect(isExperimentAlreadyViewed(orgId, experimentKey)).toBe(false);
+    });
+    it('experiment has been viewed before', () => {
+      storeExperimentParticipation(orgId, experimentKey, variationName);
+      expect(isExperimentAlreadyViewed(orgId, experimentKey)).toBe(true);
+    });
+  });
+
+  //Function used in getVariationName
+  describe('trackExperimentViewed()', () => {
+    beforeEach(() => {
+      global.AnalyticsClient = AnalyticsClient;
+    });
+
+    afterEach(() => {
+      delete global.AnalyticsClient;
+    });
+
+    it('expect trackExperimentViewed to not call trackAction as experiment viewed before', () => {
+      storeExperimentParticipation(orgId, experimentKey, variationName);
+      expect(isExperimentAlreadyViewed(orgId, experimentKey)).toBe(true);
+
+      const spy = jest.spyOn(window.AnalyticsClient, 'trackAction');
+      trackExperimentViewed(
+        orgId,
+        experimentKey,
+        ['container item'],
+        '5',
+        variationName,
+        '5',
+        '5',
+      );
+      expect(spy).not.toHaveBeenCalled();
+    });
+
+    it('expect trackExperimentViewed to have not been viewed call trackAction', () => {
+      const spy = jest.spyOn(window.AnalyticsClient, 'trackAction');
+      trackExperimentViewed(
+        orgId,
+        experimentKey,
+        ['container item'],
+        '5',
+        variationName,
+        '5',
+        '5',
+      );
+      expect(spy).toHaveBeenCalledWith('Experiment Viewed', expect.any(Object));
+    });
+  });
+
+  describe('getVariationName()', () => {
+    it('test forceAll is true return treatment', async () => {
```

Comment:
Same comment as above for test names. Here I will do:

```suggestion
    it('returns treatment when forceAll is true', async () => {
```